_pytest/test_multitenancy.py

```diff
@@ -28,110 +34,136 @@ def app(component_builder):
         'port': -1,  # unused in test app
         "backend": "mitie",
         "path": os.path.join(root_dir, "test_models"),
-        "data": os.path.join(root_dir, "data/demo-restaurants.json"),
-        "server_model_dirs": {
-            "one": "test_model_mitie",
-            "two": "test_model_mitie_sklearn",
-            "three": "test_model_spacy_sklearn",
-        }
+        "data": os.path.join(root_dir, "data/demo-restaurants.json")
     }
-    config = RasaNLUConfig(cmdline_args=_config)
-    application = create_app(config, component_builder)
-    return application
+    url = '127.0.0.1'
+    port = 5000
+    train_models(component_builder)
+    pid = os.fork()
+    if pid == 0:
+        sem.acquire()
+        config = RasaNLUConfig(cmdline_args=_config)
+        rasa = RasaNLU(config, component_builder)
+        sem.release()
+        rasa.app.run(url, port)
+        rasa.data_router.__del__()
+        os._exit(0)
+    else:
+        time.sleep(3)
+        sem.acquire()
```

Comment:
sem.acquire() makes the main process block until the http server launched by the other process is ready (as the other process is supposed to acquire the semaphore before the main one). releasing the semaphore thereafter is not necessary but you know... just to make things cleaner.